backend/src/waterfallhunter/core/entry_decision.py

```
@@ -0,0 +1,712 @@
+
+def _score_entry_components(
+    metrics: dict[str, Any], policy: EntryDecisionPolicy
+) -> tuple[dict[str, dict[str, float]], float, float, list[str], bool, float, bool, bool]:
+    components: dict[str, dict[str, float]] = {}
+    reasons: list[str] = []
+    total_points = 0.0
+    available_weight = 0.0
+
+    structure, maximum, component_reasons = _structure_points(metrics)
+    _record_component(components, "structure", structure, maximum)
+    total_points += structure; available_weight += maximum; reasons.extend(component_reasons)
```

Ruff (0.16.2)

[error] Multiple statements on one line (semicolon)

(E702)